docs/06/01/how-it-works.md

```diff
@@ -0,0 +1,73 @@
+## How the Component Adapter Works
+
+The Component Adapter uses React's Context API to provide UI components to the SDK. At its core, the system consists of:
+
+1. **ComponentsContext** - A React context that holds references to all UI components ([View on GitHub](https://github.com/Gusto/embedded-react-sdk/blob/main/src/contexts/ComponentAdapter/useComponentContext.ts))
+2. **ComponentsProvider** - A provider component that makes custom UI components available throughout the component tree ([View on GitHub](https://github.com/Gusto/embedded-react-sdk/blob/main/src/contexts/ComponentAdapter/ComponentsProvider.tsx))
+3. **useComponentContext** - A hook to access the components within SDK components ([View on GitHub](https://github.com/Gusto/embedded-react-sdk/blob/main/src/contexts/ComponentAdapter/useComponentContext.ts))
+
+When you use the SDK, all UI components (buttons, inputs, selects, etc.) are rendered through this context. By default, the SDK uses its own React Aria-based components, but you can override any or all of these with your own custom implementations.
+
+### Architecture
+
+The Component Adapter is implemented using the following architecture:
+
+```
+┌─────────────────────────┐
+│  Your Application       │
+│                         │
+│  ┌─────────────────┐    │
+│  │GustoProvider    │    │
+│  │                 │    │
+│  │ ┌─────────────┐ │    │
+│  │ │Components   │ │    │
+│  │ │Provider     │ │    │
+│  │ │             │ │    │
+│  │ │ Your Custom │ │    │
+│  │ │ Components  │ │    │
+│  │ └─────────────┘ │    │
+│  │                 │    │
+│  │ SDK Components  │    │
+│  └─────────────────┘    │
+│                         │
+└─────────────────────────┘
+```
+
+1. You create custom components that implement the required interfaces
+2. You provide these components through the `GustoProviderCustomUIAdapter` ([View on GitHub](https://github.com/Gusto/embedded-react-sdk/blob/main/src/contexts/GustoProvider/GustoProviderCustomUIAdapter.tsx))
```

Comment:
I'm wondering if we need to document the distinction between `GustoProviderCustomUIAdapter` and `GustoProvider`? components can be passed to either one. `GustoProvider` will likely be more common unless they have a real dedication to tree shaking?